src/neptune/attributes/atoms/float.py

```diff
@@ -45,5 +50,16 @@ def assign(self, value: typing.Union[FloatVal, float, int], *, wait: bool = Fals
         if not isinstance(value, FloatVal):
             value = FloatVal(value)
 
+        if is_unsupported_float(value.value):
+            warn_once(
+                message=f"WARNING: The value you're trying to log is a nonstandard float value ({str(value.value)}) "
+                f"that is not currently supported. "
+                f"We'll add support for this type of value in the future. "
+                f"For now, you can use utils.stringify_unsupported() to log one or more values as strings: "
```

Comment:
That's not true :wink:. `stringify_unsupported` doesn't help here or I don't see how it should work with float series.

Reply:
I mean, isn't it? it is applicable for atom floats. What might be a bit misleading maybe is "one or more values as strings" part, cause it is not useful in float_series. 
Is this is the part you say is not true? 😅 
(I am not the best with warning messages 😅 )

Reply:
Since I contributed to the message: My assumption was that this doesn't show for float series, but rather if the user tries to log a single value (or collection of values, hence "one or more"), in which case `stringify_unsupported` could be used.

Reply:
Keep in mind that `stringify_unsupported` means that if there will be some unsupported value (based on context as it's lazy-evaluated) it's going to be cased to a String. String may not be proper type in all of the context like -> trying to append a String value to FloatSeries.

To summarize about `stringify_unsupported`:
- It should work with atoms in most of the cases as `stringify_unsupported` is going to be either a proper value or a String; but in case there was a float value previously or something similar it may raise a MetadataInconsistency/ValueError as we're trying to do something similar to:
```python
>>> run['a'] = 4.2
>>> run['a'] = 'unsupported'
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/neptune/Repos/neptune-client/src/neptune/metadata_containers/metadata_container.py", line 105, in inner_fun
    return fun(self, *args, **kwargs)
  File "/Users/neptune/Repos/neptune-client/src/neptune/metadata_containers/metadata_container.py", line 312, in __setitem__
    self.__getitem__(key).assign(value)
  File "/Users/neptune/Repos/neptune-client/src/neptune/handler.py", line 88, in inner_fun
    return fun(self, *args, **kwargs)
  File "/Users/neptune/Repos/neptune-client/src/neptune/handler.py", line 222, in assign
    attr.process_assignment(value, wait=wait)
  File "/Users/neptune/Repos/neptune-client/src/neptune/attributes/attribute.py", line 66, in process_assignment
    return self.assign(value, wait=wait)
  File "/Users/neptune/Repos/neptune-client/src/neptune/attributes/atoms/float.py", line 51, in assign
    value = FloatVal(value)
  File "/Users/neptune/Repos/neptune-client/src/neptune/types/atoms/float.py", line 39, in __init__
    self.value = float(extract_if_stringify_value(value))
ValueError: could not convert string to float: 'unsupported'
```
but I don't have a good example yet.

- It shouldn't work for float/image series as `.extend(stringify_unsupported([5.4, 6.7, float(inf), 7.8]))` in the base case will lead to something like `.extend([5.4, 6.7, "inf", 7.8])` so it's the same as trying to append a String to a FloatSeries (should lead to an exception like MetadataInconsistency)

Reply:
Btw. Another thing that I spotted is that we're going to show double warning message with:
```python
>>> run['a'] = stringify_unsupported(float('inf'))
The value you're trying to log is an unsupported float value (inf) and will be logged as string. We'll add support for these types of values in the future.
<stdin>:1: NeptuneUnsupportedValue: WARNING: The value you're trying to log is a nonstandard float value (inf) that is not currently supported. We'll add support for this type of value in the future. For now, you can use utils.stringify_unsupported() to log one or more values as strings: run['field'] = stringify_unsupported(float(inf)
```
Is it expected?

Reply:
About the double warning -> 
true, not expected. But I was thinking, using stringify_unsupported should warn us once, so I deleted stringify_unsupported warning, now only the NeptuneUnsupportedValue will be displayed. 
I think it is a good solution for end user, @normandy7 and @Raalsky what do you think? 🤔

Reply:
So now we would have:
Scenario 1: (single float, single warning)
 `run['a'] = float("inf")`
Result 1:
`NeptuneUnsupportedValue: WARNING: The value you're trying to log is a nonstandard float value (inf) that is not currently supported. We'll add support for this type of value in the future. For now, you can use utils.stringify_unsupported() to log one or more values as strings: run['field'] = stringify_unsupported(float(inf)
  run['a'] = float("inf")`
  
Scenario 2: (float in list, warning for float in list, logging correct float)
`run['f'].extend([float("inf"), 4.0])`
Result 2:
`NeptuneUnsupportedValue: WARNING: A value you're trying to log (`inf`) will be skipped because it's a non-standard float value that is not currently supported.
  run['f'].extend([float("inf"), 4.0])`
  
 Scenario 3 (stringify unsupported):
 `run['a'] = stringify_unsupported(float("inf"))`
 Result 3:
 No warning, logged as string -> and I think it makes sense, since this is what user wanted to do in the first place.

Reply:
I rather agree with the current behavior :wink:

Reply:
For the record, I also agreed with the behavior 😄

Reply:
> For the record

@asledz Just an FYI the general process is to wait for a double approval from some dev and @normandy7 if was involved/asked for review.

Reply:
Oh sorry! Will wait in future, I just assumed a 👍🏻 emoji meant I am free to merging! 😅 Won't do that again 😇